src/node_buffer.cc

```diff
@@ -730,12 +730,20 @@ void ReadDoubleBE(const FunctionCallbackInfo<Value>& args) {
 
 
 template <typename T, enum Endianness endianness>
-uint32_t WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+void WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  bool noAssert = args[3]->BooleanValue();
+  if (!noAssert) {
+    THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
+  }
   SPREAD_ARG(args[0], ts_obj);
 
   T val = args[1]->NumberValue();
   uint32_t offset = args[2]->Uint32Value();
-  CHECK_LE(offset + sizeof(T), ts_obj_length);
+  if (!noAssert) {
+    CHECK_NOT_OOB(offset + sizeof(T) <= ts_obj_length);
+  }
```

Comment:
Drop the CHECK_LE below the conditional check. We still want to abort to prevent writing to memory bits we don't have control over.